src/libraries/System.Private.CoreLib/src/System/Number.NumberToFloatingPointBits.cs

```diff
@@ -764,5 +1552,160 @@ private static bool ShouldRoundUp(bool lsbBit, bool roundBit, bool hasTailBits)
             // two is even (this is "round-to-even").
             return roundBit && (hasTailBits || lsbBit);
         }
+
+        /// <summary> Daniel Lemire's Fast-float algorithm please refer to https://arxiv.org/abs/2101.11408 </summary>
+        internal static (int power2, ulong mantissa) ComputeFloat(long q, ulong w, FloatingPointInfo info)
```

Comment:
`power2` should likely just be named `exponent` since it represents the "exponent" bits and is the term used everywhere else.

Likewise, `q` and `w` aren't very descriptive here. It would be good to at least give a brief comment explaining what they are if the intent is to match the paper.

Reply:
Also, nit: The convention is that tuple members are `PascalCase` since they represent public struct members and would therefore be equivalently cased in an explicit named struct.

Reply:
switched to pascal case
yes intent is to match the paper, I'll add a description as requested.

Reply:
I also renamed power2 as exponent.